CMakeLists.txt

```diff
@@ -7,69 +7,94 @@
 #  License: BSD License ; see LICENSE
 #
 
-
-cmake_minimum_required( VERSION 3.4.3 )
+cmake_minimum_required( VERSION 3.12 )
 
 project( vtu11 CXX )
 
-set( CMAKE_CXX_STANDARD 11 )
+# --------------- setup vtu11 interface library ----------------
+
+add_library( vtu11 INTERFACE )
+add_library( vtu11::vtu11 ALIAS vtu11 )
+
+target_include_directories( vtu11 INTERFACE . )
+target_compile_features( vtu11 INTERFACE cxx_std_11 )
+
+find_package( ZLIB )
+
+if( ZLIB_FOUND )
+
+    message( STATUS "Enabling vtu11 with zlib compression" )
 
-include(CTest)
+    target_include_directories( vtu11 INTERFACE ${ZLIB_INCLUDE_DIRS} )
+    target_link_libraries( vtu11 INTERFACE ${ZLIB_LIBRARIES} )
+    target_compile_definitions( vtu11 INTERFACE VTU11_ENABLE_ZLIB )
 
-OPTION ( VTU11_STRICT_COMPILER "Enable more warnings as errors." OFF )
+endif( ZLIB_FOUND )
 
-# If compiler is g++: Enable further  warnings and treat all warnings as errors
-if( VTU11_STRICT_COMPILER )
+# ------------------- setup vtu11 unit tests -------------------
+
+option( VTU11_ENABLE_TESTS "Build vtu11 unit tests." OFF )
+
+if( ${VTU11_ENABLE_TESTS} )
+     
+    set( VTU11_HEADERS 
+         vtu11/vtu11.hpp 
+         vtu11/inc/alias.hpp 
+         vtu11/inc/filesystem.hpp 
+         vtu11/inc/utilities.hpp 
+         vtu11/inc/writer.hpp 
+         vtu11/inc/zlibWriter.hpp 
+         vtu11/impl/utilities_impl.hpp 
+         vtu11/impl/vtu11_impl.hpp 
+         vtu11/impl/writer_impl.hpp 
+         vtu11/impl/zlibWriter_impl.hpp )
+         
+    set( VTU11_TEST_SOURCES 
+         test/main_test.cpp 
+         test/pwrite_pyramids3D_test.cpp 
+         test/utilities_test.cpp 
+         test/vtu11_testing.cpp 
+         test/vtu11_testing.hpp 
+         test/write_hexahedras3D_test.cpp 
+         test/write_icosahedron3D_test.cpp 
+         test/write_pyramids3D_test.cpp 
+         test/write_square2D_test.cpp )
```

Comment:
why no `GLOB`?

Reply:
In my code I added a `files.cmake` in every source folder which only sets one variable with all file names.

Reply:
Like here: https://gitlab.com/phmkopp/mlhp/-/blob/master/source/core/files.cmake